---
title: "B1: The Cancer Genome Atlas in Bioconductor -- focus on mutations"
author: "Vincent J. Carey, stvjc at channing.harvard.edu"
date: "`r format(Sys.time(), '%B %d, %Y')`"
vignette: >
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteIndexEntry{B1: The Cancer Genome Atlas in Bioconductor -- focus on mutations}
  %\VignetteEncoding{UTF-8}
output:
  BiocStyle::html_document:
    highlight: pygments
    number_sections: yes
    theme: united
    toc: yes
---

```{r setup,message=FALSE,echo=FALSE}
library(BiocStyle)
library(BiocOV)
library(MultiAssayExperiment)
library(ggplot2)
library(plotly)
library(EnsDb.Hsapiens.v75)
library(ensembldb)
```

# Introduction

[The Cancer Genome Atlas (TCGA)](https://portal.gdc.cancer.gov/) collects information on thousands
of cases of tumors from 68 anatomical sites.  Several approaches to working
with TCGA are available in Bioconductor.  We'll focus on the \Biocpkg{curatedTCGAData}
package developed by the Waldron Lab at City University of New York.

```{r lkcur,message=FALSE}
library(curatedTCGAData)
```

The following excerpt from the manual page for the `curatedTCGAData`
function defines the scope of genomic assays available for most tumor types.
```
Available Assays:

     Below is a list of partial ExperimentList assay names and their
     respective description. These assays can be entered as part of the
     'assays' argument in the main function. Partial glob matches are
     allowed such as: ''CN*'' for "CNASeq", "CNASNP", "CNVSNP" assays.
     Credit: Ludwig G.

     ExperimentList data types   Description
     ----------------------------------------------------------------------------
     SummarizedExperiment*
       RNASeqGene                Gene expression values
       RNASeq2Gene               RSEM TPM gene expression values
       RNASeq2GeneNorm           Upper quartile normalized RSEM TPM gene
                                 expression values
       miRNAArray                Probe-level  miRNA expression values
       miRNASeqGene              Gene-level log2 RPM miRNA expression values
       mRNAArray                 Unified gene-level mRNA expression values
       mRNAArray_huex            Gene-level mRNA expression values from Affymetrix
                                 Human Exon Array
       mRNAArray_TX_g4502a       Gene-level mRNA expression values from Agilent
                                 244K Array
       mRNAArray_TX_ht_hg_u133a  Gene-level mRNA expression values from Affymetrix
                                 Human Genome U133 Array
       GISTIC_AllByGene          Gene-level GISTIC2 copy number values
       GISTIC_ThresholdedByGene  Gene-level GISTIC2 thresholded discrete copy
                                 number values
       RPPAArray                 Reverse Phase Protein Array normalized protein
                                 expression values
     RangedSummarizedExperiment
       GISTIC_Peaks              GISTIC2 thresholded discrete copy number values
                                 in recurrent peak regions
     SummarizedExperiment with HDF5Array DelayedMatrix
       Methylation_methyl27      Probe-level methylation beta values from Illumina
                                 HumanMethylation 27K BeadChip
       Methylation_methyl450     Probe-level methylation beta values from Infinium
                                 HumanMethylation 450K BeadChip
     RaggedExperiment
       CNASNP                    Segmented somatic Copy Number Alteration calls
                                 from SNP array
       CNVSNP                    Segmented germline Copy Number Variant calls from
                                 SNP Array
       CNASeq                    Segmented somatic Copy Number Alteration calls
                                 from low pass DNA Sequencing
       Mutation*                 Somatic mutations calls
       CNACGH_CGH_hg_244a        Segmented somatic Copy Number Alteration calls
                                 from CGH Agilent Microarray 244A
       CNACGH_CGH_hg_415k_g4124a Segmented somatic Copy Number Alteration calls
                                 from CGH Agilent Microarray 415K
     * All can be converted to RangedSummarizedExperiment (except RPPAArray) with
     TCGAutils
     
version:

     The new version 2.0.1 includes various improvements including an
     additional assay that provides 'RNASeq2Gene' data as RSEM TPM gene
     expression values (issue #38). Additional changes include genomic
```

# Mutations in the BRCA collection

```{r lkbrca,message=FALSE}
library(curatedTCGAData)
library(TCGAutils)
br = curatedTCGAData("BRCA", "Mutation", version="2.0.1", dry.run=FALSE)
brp = TCGAprimaryTumors(br)  # remove some normal tissue samples
mut = experiments(brp)[[1]]
mut
```

`mut` is an instance of the RaggedExperiment class.  This
class was devised to accommodate data in which there is
wide variation in the number of features recorded on
each patient.  RaggedExperiment is also unusual in that
there are many assay components.

```{r lkan}
length(assayNames(mut))
head(assayNames(mut),10)
```
## Distributions of mutation counts

This computation is unwieldy.  We just
record the gene names and number of times
some mutation is reported for some sample.

```
tail(sort(table(as.character(assay(mut)))),20)

   SPEN   FCGBP    RYR3     DMD CROCCP2   HMCN1     DST     FLG   SYNE1    RYR2 
     61      63      63      64      66      67      69      69      74      76 
   MLL3    MUC4   MUC12   GATA3  MAP3K1    CDH1   MUC16    TP53     TTN  PIK3CA 
     92      92     101     106     106     123     152     332     360     384 

```

The distribution of counts of mutations per individual:
```{r indiv}
nmut_per_sample = apply(assay(mut),2,function(x)sum(!is.na(x)))
summary(nmut_per_sample)
```

## Identifying hypermutated samples

A hypermutator:
```{r lkhy}
which.max(nmut_per_sample)
table(table(assay(mut[,285])))
which(table(assay(mut[,285]))==36)
```

Several:
```{r lkhy2}
hyp = tail(sort(nmut_per_sample))
himut = mut[,names(hyp)]
himut
```
This individual presents single mutations in many genes.
```{r lkhi1}
table(table(assay(himut[,1])))
```
This one has 4 genes with more than 10 mutations in each.
```{r lkhi2}
table(table(assay(himut[,4])))
```

## Mutation types: Insertion, deletion, ...

We include some helper code here to produce comparisons
of distributions of mutation types in genes mutated
in breast cancer.

```{r lkstr, message=FALSE}
library(MultiAssayExperiment)
cropSE = function (se, sym = "AHNAK", 
  base = ensembldb::genes(EnsDb.Hsapiens.v75::EnsDb.Hsapiens.v75), gr=NULL) {
    rr = rowRanges(se)
    if (is.null(gr)) cr = base[which(base$symbol == sym)]
    else cr = gr
    fi = findOverlaps(rr, cr, ignore.strand = TRUE)
    se[queryHits(fi), ]
}

.muttab = function(se, sym="AHNAK", 
   base = ensembldb::genes(EnsDb.Hsapiens.v75::EnsDb.Hsapiens.v75),
   assayind=5) {
 cr = cropSE(se, sym, base)
 tab = table(assay(cr,assayind))
 data.frame(gene=sym, muttype=names(tab), freq=as.numeric(tab))
}

muttab = function(se, sym=c("MAP3K1", "CDH1", "MUC16", "TP53", "TTN", "PIK3CA"), 
   base = ensembldb::genes(EnsDb.Hsapiens.v75::EnsDb.Hsapiens.v75),
   assayind=5) {
  ans = lapply(sym, function(x) .muttab(se, x, base, assayind))
  do.call(rbind, ans)
}
```

Here we produce an interactive plot.  You can hover over the bars
to see text on type and count for each mutation type for each gene.

```{r dopl, message=FALSE}
dd = muttab(mut)
library(ggplot2)
p = ggplot(dd, aes(fill=muttype, y=freq, x=gene)) + 
   geom_bar(stat="identity", position="dodge")
library(plotly)
ggplotly(p)
```

## Identifying a mutated exon

We limit attention to the mutations that
have been annotated to PIK3CA.
```{r docrpi}
pi = cropSE(mut, "PIK3CA")
pi
```
The catalog of all exons:
```{r lkex}
ex = exons(EnsDb.Hsapiens.v75)
ex
```

The findOverlaps methods work with a "query" and
a "subject", and produce a "hits" object, which
has methods "queryHits" and "subjectHits".

```{r dofi}
fo = findOverlaps(ex, rowRanges(pi)) 
table(queryHits(fo))
```

Here's an exon within PIK3CA within which many mutations were observed:
```{r lkhiex}
ex[483987]
lit = cropSE(mut, gr=ex[483987])
table(start(rowRanges(lit)))
```
Several positions are found recurrently mutated.

## Exercise

What is the most commonly mutated position in the breast cancer genomes
in TCGA?

# Mutation in relation to survival times

## Exercise

Use `example(build_surv_for_mut)` to compare survival
profiles for individuals with and without TTN mutation.

Produce a similar display comparing individuals with and
without PIK3CA mutation.

